src/main/java/org/folio/circulation/domain/notes/NoteCreator.java

```diff
@@ -0,0 +1,31 @@
+package org.folio.circulation.domain.notes;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.domain.NoteLink;
+import org.folio.circulation.domain.validation.GeneralNoteTypeValidator;
+import org.folio.circulation.infrastructure.storage.notes.NotesRepository;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.utils.CollectionUtil;
+
+public class NoteCreator {
+  private final NotesRepository notesRepository;
+
+  public NoteCreator(NotesRepository notesRepository) {
+    this.notesRepository = notesRepository;
+  }
+
+  public CompletableFuture<Result<Note>> createNote(String userId, String message) {
```

Comment:
Maybe something like:
```suggestion
  public CompletableFuture<Result<Note>> createGeneralUserNote(String userId, String message) {
```

Reply:
I'm happy to change this. I'd like to understand a little more about why the more specific name is preferable? 

Is it important to the client that the note is of a specific type? 

Is it to create space for other kinds of notes (I'm not aware of any anticipated at the moment)?

Reply:
Yes, since the method has harcoded type (which I believe is important info for a note), I think it make sense to reflect it in the name.

Reply:
Ok, I didn't think that detail was relevant at that level of the process. Thank you for explaining your reasoning. I'll make the change.